src/main/java/com/uid2/shared/attest/AttestationToken.java

```diff
@@ -50,26 +49,67 @@ public static AttestationToken fromPlaintext(String plaintext) {
     public static AttestationToken fromEncrypted(String encryptedToken, String paraphrase, String salt) {
         try {
             String[] parts = encryptedToken.split("-");
-            if(parts.length != 2) {
-                throw new Exception("token must satisfy format: <Base64>-<Base64>");
+            if (parts.length == 2) {
+                String plainText = decryptOld(
+                        Base64.getDecoder().decode(parts[0]),
+                        Base64.getDecoder().decode(parts[1]),
+                        paraphrase, salt);
+                return fromPlaintext(plainText);
+            } else if (parts.length == 3) {
+                if (!parts[2].equals("g")) {
+                    throw new Exception("invalid attestation token: invalid encryption algorithm");
+                }
+                String plainText = decrypt(
+                        Base64.getDecoder().decode(parts[0]),
+                        Base64.getDecoder().decode(parts[1]),
+                        paraphrase, salt);
+                return fromPlaintext(plainText);
+            } else {
+                throw new Exception("invalid attestation token format");
             }
-
-            byte[] cipherText = Base64.getDecoder().decode(parts[0]);
-            byte[] iv = Base64.getDecoder().decode(parts[1]);
-            Cipher cipher = Cipher.getInstance(Algorithm);
-            cipher.init(Cipher.DECRYPT_MODE, getKeyFromPassword(paraphrase, salt), new IvParameterSpec(iv));
-            byte[] plaintext = cipher.doFinal(cipherText);
-            return fromPlaintext(new String(plaintext));
         } catch (Exception e) {
             LOGGER.debug("failed to decrypt attestation token: {}", e.getMessage());
             return AttestationToken.Failed();
         }
     }
 
+    private static String decrypt(byte[] cipherText, byte[] iv, String paraphrase, String salt) throws Exception {
+        Cipher cipher = Cipher.getInstance("AES/GCM/NoPadding");
+        cipher.init(Cipher.DECRYPT_MODE, getKeyFromPassword(paraphrase, salt), new GCMParameterSpec(128, iv));
+        byte[] plaintext = cipher.doFinal(cipherText);
+        return new String(plaintext);
+    }
+
+    @Deprecated
+    private static String decryptOld(byte[] cipherText, byte[] iv, String paraphrase, String salt) throws Exception {
+        Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
+        cipher.init(Cipher.DECRYPT_MODE, getKeyFromPassword(paraphrase, salt), new IvParameterSpec(iv));
+        byte[] plaintext = cipher.doFinal(cipherText);
+        return new String(plaintext);
+    }
+
+    // TODO: replace encode with encodeNew
```

Comment:
Is the TODO still necessary?

Reply:
Yes, there will be a follow-up PR
- enabling GCM as the encrypting method
- removing old methods